BaseTools/Source/C/VfrCompile/Pccts/dlg/dlg.1
.TH dlg 1 "April 1994" "DLG" "PCCTS Manual Pages"
.SH NAME
dlg \- DFA Lexical Analyzer Generator
.SH SYNTAX
.LP
\fBdlg\fR [\fIoptions\fR] \fIlexical_spec\fR [\fIoutput_file\fR]
.SH DESCRIPTION
.B dlg
is a tool that produces fast deterministic finite automata for recognizing
regular expressions in input.
.SH OPTIONS
.IP "\fB-CC\fR"
Generate C++ output.  The \fIoutput_file\fP is not specified in this
case.
.IP "\fB-C\fR[\fP level\fR]
Where \fPlevel\fR is the compression level used.  0 indications no
compression, 1 removes all unused characters from the transition from table,
and 2 maps equivalent characters into the same character classes.  It is
suggested that level -C2 is used, since it will significantly reduce the size
of the dfa produced for lexical analyzer.
.IP "\fB-m\fP
Produces the header file for the lexical mode with a name other than
the default name of "mode.h".
.IP \fB-i\fP
An interactive, or as interactive as possible, parser is produced.  A character
is only obtained when required to decide which state to go to.  Some care
must be taken to obtain accept states that do not require look ahead at the
next character to determine if that is the stop state.  Any regular expression
with a Kleene closure at the end is guaranteed to require another character
of look ahead.
.IP "\fB-cl\fP class
Specify a class name for DLG to generate.  The default is DLGLexer.
'class' will be a subclass of DLGLexerBase; only used for -CC.
.IP \fB-ci\fP
The automaton will treat upper and lower case characters identically.
This is accomplished in the automaton; the characters in the lexical
buffer are unmodified.
.IP \fB-cs\fP
Upper and lower case characters are treated as distinct.  This is the
default.
.IP "\fB-o\fP dir
Directory where output files should go (default=".").  This is very
nice for keeping the source directory clear of ANTLR and DLG spawn.
.IP \fB-Wambiguity\fP
Warns if more than one regular expression could match the same character
sequence.  The warnings give the numbers of the expressions in the dlg
lexical specification file.  The numbering of the expressions starts at one.
Multiple warnings may be print for the same expressions.
.IP \-
Used in place of file names to get input from standard in or send output
to standard out.
.SH "SPECIAL CONSIDERATIONS"
.PP
\fIDlg\fP works...  we think.  There is no implicit guarantee of
anything.  We reserve no \fBlegal\fP rights to the software known as
the Purdue Compiler Construction Tool Set (PCCTS) \(em PCCTS is in the
public domain.  An individual or company may do whatever they wish
with source code distributed with PCCTS or the code generated by
PCCTS, including the incorporation of PCCTS, or its output, into
commercial software.  We encourage users to develop software with
PCCTS.  However, we do ask that credit is given to us for developing
PCCTS.  By "credit", we mean that if you incorporate our source code
into one of your programs (commercial product, research project, or
otherwise) that you acknowledge this fact somewhere in the
documentation, research report, etc...  If you like PCCTS and have
developed a nice tool with the output, please mention that you
developed it using PCCTS.  As long as these guidelines are followed, we
expect to continue enhancing this system and expect to make other
tools available as they are completed.
.SH FILES
.B mode.h
,
.B dlgauto.h
,
.B dlgdef.h
.SH SEE ALSO
.BR antlr (1),
.BR pccts (1)
.SH BUGS


BaseTools/Source/C/VfrCompile/Pccts/dlg/dlg.r
/*
	File:		dlgMPW.r
	Target:		dlg 133MR
	Created:    Monday, June 15, 1998 4:44:11 AM
	Author:		Kenji Tanaka (kentar@osa.att.ne.jp)
*/

#include "cmdo.r"

resource 'cmdo' (128, "Dlg") {
	{	/* array dialogs: 1 elements */
		/* [1] */
		295,
		"DLG -- Purdue Compiler Construction Tool"
		" Set (PCCTS) lexical analyzer generator.",
		{	/* array itemArray: 18 elements */
			/* [1] */
			NotDependent {

			},
			CheckOption {
				NotSet,
				{35, 175, 50, 225},
				"On",
				"-CC",
				"When this control is checked, DLG genera"
				"tes a scanner using C++ classes rather t"
				"han C functions."
			},
			/* [2] */
			Or {
				{	/* array OrArray: 1 elements */
					/* [1] */
					1
				}
			},
			RegularEntry {
				"Lexer Class Name:",
				{35, 225, 50, 355},
				{35, 355, 51, 450},
				"DLGLexer",
				keepCase,
				"-cl",
				"This entry specifies the name DLG uses f"
				"or the C++ lexer class."
			},
			/* [3] */
			NotDependent {

			},
			TextBox {
				gray,
				{25, 165, 60, 460},
				"C++ Code Generation"
			},
			/* [4] */
			NotDependent {

			},
			Files {
				InputFile,
				RequiredFile {
					{37, 25, 56, 135},
					"Input File",
					"",
					"Choose the lexical description file for "
					"DLG to process."
				},
				Additional {
					"",
					"",
					"",
					"",
					{	/* array TypesArray: 1 elements */
						/* [1] */
						text
					}
				}
			},
			/* [5] */
			Or {
				{	/* array OrArray: 1 elements */
					/* [1] */
					-1
				}
			},
			Files {
				OutputFile,
				RequiredFile {
					{66, 25, 85, 135},
					"Output File",
					"",
					"Choose the name of the file that will ho"
					"ld the DLG-produced scanner."
				},
				NoMore {

				}
			},
			/* [6] */
			Or {
				{	/* array OrArray: 2 elements */
					/* [1] */
					1,
					/* [2] */
					5
				}
			},
			Dummy {

			},
			/* [7] */
			NotDependent {

			},
			Redirection {
				DiagnosticOutput,
				{90, 25}
			},
			/* [8] */
			NotDependent {

			},
			TextBox {
				gray,
				{25, 20, 132, 145},
				"Files"
			},
			/* [9] */
			NotDependent {

			},
			Files {
				DirOnly,
				OptionalFile {
					{68, 175, 84, 305},
					{88, 175, 107, 305},
					"Output Directory",
					":",
					"-o",
					"",
					"Choose the directory where DLG will put "
					"its output.",
					dim,
					"Output DirectoryI",
					"",
					""
				},
				NoMore {

				}
			},
			/* [10] */
			NotDependent {

			},
			RegularEntry {
				"Mode File Name:",
				{68, 315, 83, 450},
				{88, 315, 104, 450},
				"mode.h",
				keepCase,
				"-m",
				"This entry specifies the name DLG uses f"
				"or its lexical mode output file."
			},
			/* [11] */
			NotDependent {

			},
			RadioButtons {
				{	/* array radioArray: 3 elements */
					/* [1] */
					{134, 175, 149, 255}, "None", "", Set, "When this option is selected, DLG will n"
					"ot compress its tables.",
					/* [2] */
					{134, 265, 149, 345}, "Level 1", "-C1", NotSet, "When this option is selected, DLG will r"
					"emove all unused characters from the tra"
					"nsition-from table.",
					/* [3] */
					{134, 360, 149, 450}, "Level 2", "-C2", NotSet, "When this option is selected, DLG will p"
					"erform level 1 compression plus it will "
					"map equivalent characters into the same "
					"character classes."
				}
			},
			/* [12] */
			NotDependent {

			},
			TextBox {
				gray,
				{124, 165, 156, 460},
				"Table Compression"
			},
			/* [13] */
			NotDependent {

			},
			CheckOption {
				Set,
				{165, 20, 180, 145},
				"Case Sensitive",
				"-ci",
				"When this control is checked, the DLG au"
				"tomaton will treat upper and lower case "
				"characters identically."
			},
			/* [14] */
			NotDependent {

			},
			CheckOption {
				NotSet,
				{165, 150, 180, 300},
				"Interactive Scanner",
				"-i",
				"When this control is checked, DLG will g"
				"enerate as interactive a scanner as poss"
				"ible."
			},
			/* [15] */
			NotDependent {

			},
			CheckOption {
				NotSet,
				{165, 310, 180, 460},
				"Ambiguity Warnings",
				"-Wambiguity",
				"When this control is checked, DLG warns "
				"if more than one regular expression coul"
				"d match the same character sequence."
			},
			/* [16] */
			NotDependent {

			},
			VersionDialog {
				VersionString {
					"1.33MR"
				},
				"PCCTS was written by Terence Parr, Russe"
				"ll Quong, Will Cohen, and Hank Dietz: 19"
				"89-1998. MPW port by Scott Haney.",
				noDialog
			},
			/* [17] */
			And {
				{	/* array AndArray: 2 elements */
					/* [1] */
					4,
					/* [2] */
					6
				}
			},
			DoItButton {

			},
			/* [18] */
			NotDependent {

			},
			CheckOption {
				NotSet,
				{142, 20, 157, 148},
				"Generate ANSI C",
				"-ga",
				"When this control is checked, DLG genera"
				"tes ANSI C compatible code."
			}
		}
	}
};

